### PR TITLE
Adding youtube card

### PR DESCRIPTION
**Pull request by bnookala**
Looks like this: 
youtube-card

And uses the embeddable player with an iframe to drop it in: https://developers.google.com/youtube/player_parameters#Overview

### CODE REVIEW

File: src/Attachment.tsx [hunk 1]
```diff
@@ -13,6 +15,50 @@ const buttons = (
         { buttons.map((button, index) => <li key={ index }><button onClick={ () => onClickButton(button.type, button.value) }>{ button.title }</button></li>) }
     </ul>;
 
+const isYoutube = (src: string) => {
+    let youtubeUrl = new URI(src);
```
Comment (billba) [inline src/Attachment.tsx]: Our convention in WebChat is to use `const` when possible. Please humor me here and below.

File: src/Attachment.tsx [hunk 2]
```diff
@@ -13,6 +15,50 @@ const buttons = (
...
+                    .domain("youtube.com")
+                    .port("")
+                    .segment(["embed", embedId])
+                    .search(embedUrlQuery)
+                    .toString();
+
+    return <iframe
+        type="text/html"
+        src={ embedUrl } >
+    </iframe>
```
Comment (billba) [inline src/Attachment.tsx]: I like my tags to line up. Will you either put this all on one line, or do:

```
return (
    <iframe
        ...
    </iframe>
);

File: src/Attachment.tsx [hunk 3]
```diff
@@ -13,6 +15,50 @@ const buttons = (
...
+
+    if ((youtubeUrl.domain() !== 'youtube.com')) {
+        return false;
+    }
+
+    if (embedId === undefined) {
+        return false;
+    }
+
+    return true;
```
Comment (billba) [inline src/Attachment.tsx]: This is too wordy for my taste. How about we collapse this all into:

```
return youtubeUrl.domain() === "youtube.com" && youtubeUrl.search(true).v !== undefined;
```

File: src/Attachment.tsx [hunk 4]
```diff
@@ -25,6 +71,10 @@ const Media = (props: {
     const { type, ... mediaProps } = props; // this allows us to keep 'type' out of the final HTML
     switch (type) {
         case 'video':
+            if (isYoutube(mediaProps.src)) {
+                return <Youtube { ... mediaProps } />
+            }
+
```
Comment (billba) [inline src/Attachment.tsx]: It bugs me that we end up calling URI twice on the same string. And if we add future special cases, we'll do it more.

How about we turn this into a function called Video which will look at the URI for each special-case domain in turn, returning the appropriate `<iframe>` as appropriate, or returning a `<video>` tag if none apply. All that logic can be in one function.

File: botchat.css [hunk 5]
```diff
@@ -149,6 +149,10 @@
         max-width: 100%;
     }
 
+    .wc-message-content .video iframe {
+        border: 0px;
```
Comment (danmarshall) [inline botchat.css]: 0 is ok, without the px units

**Comment by danmarshall** [comment]
we don't necessarily need to include another library - have a look at this: http://stackoverflow.com/questions/3213531/creating-a-new-location-object-in-javascript

File: src/Attachment.tsx [hunk 6]
```diff
@@ -13,6 +15,50 @@ const buttons = (
         { buttons.map((button, index) => <li key={ index }><button onClick={ () => onClickButton(button.type, button.value) }>{ button.title }</button></li>) }
     </ul>;
 
+const isYoutube = (src: string) => {
+    let youtubeUrl = new URI(src);
+    let queryParams = youtubeUrl.search(true);
+    let embedId = queryParams.v;
+
+    if ((youtubeUrl.domain() !== 'youtube.com')) {
```
Comment (danmarshall) [inline src/Attachment.tsx]: Youtube also uses their own short url format http://youtu.be

File: src/Attachment.tsx (hunk 3 above)
Comment (danmarshall) [inline src/Attachment.tsx]: also, why compare so strictly against ```undefined``` - we can also just say ```!embedId```

**Comment by danmarshall** [comment]
@jecristi can you review the visual design of this? It may be that we also want to display the URL in addition to the player.

**Comment by danmarshall** [comment]
@bnookala nice work, thanks!

File: src/Attachment.tsx (hunk 4 above)
Comment (danmarshall) [inline src/Attachment.tsx]: I tend to agree, since it may become a verbose block:
```
 if (isYouTube...)
   return <YouTube...>
 if (isVimeo...)
   return <Vimeo...>
 if (isFacebook...)
   return <Facebook...>
 if (isInstagram...)
   return <Instagram...>
```

File: src/Attachment.tsx (hunk 1 above)
Comment (bnookala) [inline src/Attachment.tsx]: 👍

File: src/Attachment.tsx (hunk 3 above)
Comment (bnookala) [inline src/Attachment.tsx]: 👍

File: src/Attachment.tsx (hunk 2 above)
Comment (bnookala) [inline src/Attachment.tsx]: 👍

File: botchat.css (hunk 5 above)
Comment (bnookala) [inline botchat.css]: 👍

File: src/Attachment.tsx (hunk 4 above)
Comment (billba) [inline src/Attachment.tsx]: @danmarshall You really get me.

File: src/Attachment.tsx (hunk 6 above)
Comment (bnookala) [inline src/Attachment.tsx]: 👍

File: src/Attachment.tsx (hunk 4 above)
Comment (bnookala) [inline src/Attachment.tsx]: 👍

**Comment by bnookala** [comment]
@danmarshall, @billba once more for good measure, incorporating your suggestions.

@danmarshall I'm a bit partial to using the URI.js library it provides many convenience methods to things a`Location` object doesn't: ie: building/decomposing the querystring from or into/an object, building up a directory structures, etc. I imagine as we think about adding more embeddable attachments, it will prove to be quite useful.

**Comment by billba** [comment]
This turned out great, thanks!
